File: core-service/src/main/java/com/sandjelkovic/dispatchd/domain/data/entity/ImportStatus.java
package com.sandjelkovic.dispatchd.domain.data.entity;

import org.hibernate.validator.constraints.Length;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.NamedQuery;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.time.ZonedDateTime;


@Entity
@Table(name = "ImportStatus")
@NamedQuery(name = "ImportStatus.findAll", query = "SELECT s FROM ImportStatus s")
public class ImportStatus extends BasicEntity implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	@Column(unique = true, nullable = false)
	private Long id;

	@Column(nullable = true, length = 500)
	//@URL
	@Length(max = 1000)
	private String mediaUrl;

	@Column(nullable = false)
	@NotNull
	private ZonedDateTime initiationTime;

	@Column(nullable = true)
	private ZonedDateTime finishTime;

	@Column(nullable = false)
	@Enumerated(EnumType.STRING)
	private ImportProgressStatus status;

	public ImportStatus() {
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getMediaUrl() {
		return mediaUrl;
	}

	public void setMediaUrl(String mediaUrl) {
		this.mediaUrl = mediaUrl;
	}

	public ZonedDateTime getInitiationTime() {
		return initiationTime;
	}

	public void setInitiationTime(ZonedDateTime initiationTime) {
		this.initiationTime = initiationTime;
	}

	public ZonedDateTime getFinishTime() {
		return finishTime;
	}

	public void setFinishTime(ZonedDateTime finishTime) {
		this.finishTime = finishTime;
	}

	public ImportProgressStatus getStatus() {
		return status;
	}

	public void setStatus(ImportProgressStatus status) {
		this.status = status;
	}

	public ImportStatus id(final Long id) {
		this.id = id;
		return this;
	}

	public ImportStatus mediaUrl(final String mediaUrl) {
		this.mediaUrl = mediaUrl;
		return this;
	}

	public ImportStatus initiationTime(final ZonedDateTime initiationTime) {
		this.initiationTime = initiationTime;
		return this;
	}

	public ImportStatus finishTime(final ZonedDateTime finishTime) {
		this.finishTime = finishTime;
		return this;
	}

	public ImportStatus status(final ImportProgressStatus status) {
		this.status = status;
		return this;
	}

	@Override
	protected Object getInternalId() {
		return getId();
	}
}

File: content-service/src/main/kotlin/com/sandjelkovic/dispatchd/content/data/entity/ImportStatus.kt
package com.sandjelkovic.dispatchd.content.data.entity

import org.hibernate.validator.constraints.Length
import java.time.ZonedDateTime
import javax.persistence.*

/**
 * @author sandjelkovic
 * @date 21.4.18.
 */
@Entity
data class ImportStatus(
    @Id
    @GeneratedValue
    var id: Long? = null,
    @Column(nullable = false, length = 1000)
    @Length(max = 1000)
    var mediaUrl: String = "",

    @Column(nullable = false)
    var initiationTime: ZonedDateTime = ZonedDateTime.now(),

    var finishTime: ZonedDateTime? = null,

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    var status: ImportProgressStatus = ImportProgressStatus.QUEUED
)

enum class ImportProgressStatus(val field: String) {
    QUEUED("QUEUED"),
    IN_PROGRESS("IN_PROGRESS"),
    SUCCESS("SUCCESS"),
    ERROR_SHOW_ALREADY_EXISTS("ERROR_SHOW_ALREADY_EXISTS"),
    ERROR_REMOTE_SERVER("ERROR_REMOTE_SERVER"),
    ERROR("ERROR");
}

inline class ImportStatusId(val value: Long)
